snap/snapcraft.yaml

```diff
@@ -0,0 +1,61 @@
+name: pebble
+adopt-info: pebble-release-data
+summary: Pebble - take control of your internal daemons!
+description: |
+  Pebble helps you to orchestrate a set of local service processes
+  as an organized set. It resembles well-known tools such as supervisord,
+  runit, or s6, in that it can easily manage non-system processes
+  independently from the system services, but it was designed with unique
+  features that help with more specific use cases.
+
+  **Usage**
+
+  * The Pebble snap's default directory is `$SNAP_USER_DATA` (which translates
+  to `$HOME/snap/pebble/<rev>/`).
+
+  **Documentation**
+
+  To learn more about Pebble please check the project's documentation at
+  https://github.com/canonical/pebble.
+issues: https://github.com/canonical/pebble/issues
+source-code: https://github.com/canonical/pebble
+license: GPL-3.0
+base: bare
+build-base: core22
+confinement: classic
+
+apps:
+  pebble:
+    command: pebble-wrapper
+
+parts:
+  pebble:
+    plugin: go
+    source: .
+    build-snaps:
+      - go
+    build-environment:
+      - CGO_ENABLED: 0
+      - GOFLAGS: -trimpath -ldflags=-w -ldflags=-s
+    override-build: |
+      go generate ./cmd
+      craftctl default
+
+  pebble-release-data:
+    plugin: nil
+    source: .
+    override-build: |
+      version="$(${CRAFT_STAGE}/bin/pebble version --client)"
+
+      # Set snap version
+      snapcraftctl set-version "$version"
```

Comment:
Is there a reason you prefer using snapcraftctl here (and further down)?

```craftctl set version="$version"```

Reply:
thanks for noticing @flotter . `craftctl` is indeed preferred. I've changed it.